### PR TITLE
feat(launch): Fail queued runs items that fail to start

### PR DESCRIPTION
**Pull request by KyleGoyette**
[Fixes WB-12590](https://wandb.atlassian.net/browse/WB-12590)
Fixes #NNNN

Description
-----------
- Uses new mutation FailQunQueueItem to set run queue items that never created a wandb run to the FAILED state.

Testing
-------
Locally using normal launch jobs and sweeps on launch

Checklist
-------
- [ x] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: tests/pytest_tests/unit_tests/test_launch/test_agent/test_agent.py [hunk 1]
```diff
@@ -69,3 +70,103 @@ def test_max_scheduler_setup_fail(mocker, num_schedulers):
...
+    job = JobAndRunStatus("run_queue_item_id")
+    job.run_id = "test_run_id"
+    job.project = MagicMock()
+    agent._jobs = {"thread_1": job}
+    agent.finish_thread_id("thread_1")
+    assert len(agent._jobs) == 0
+    assert not mocker.api.fail_run_queue_item.called
+
+
+def test_thread_finish_sweep_fail(mocker):
```
Comment (gtarpenning) [inline tests/pytest_tests/unit_tests/test_launch/test_agent/test_agent.py:104]: sick

File: wandb/sdk/internal/internal_api.py [hunk 2]
```diff
@@ -409,12 +410,15 @@ def parse_slug(
...
+    def server_info_introspection(self) -> Tuple[List[str], List[str], List[str]]:
         query_string = """
            query ProbeServerCapabilities {
                QueryType: __type(name: "Query") {
                    ...fieldData
                 }
+                MutationType: __type(name: "Mutation") {
+                     ...fieldData
+                }
                ServerInfoType: __type(name: "ServerInfo") {
```
Comment (gtarpenning) [inline wandb/sdk/internal/internal_api.py:422]: nit: whitespace is off here

File: wandb/sdk/launch/agent/agent.py [hunk 3]
```diff
@@ -233,9 +251,24 @@ def update_status(self, status: str) -> None:
...
+            try:
+                run_info = self._api.get_run_info(
+                    self._entity, job_and_run_status.project, job_and_run_status.run_id
+                )
+                # sweep runs exist but have no info before they are started
+                if run_info is None:
+                    self.fail_run_queue_item(job_and_run_status.run_queue_item_id)
+            # normal runs just have no info
+            except CommError:
+                self.fail_run_queue_item(job_and_run_status.run_queue_item_id)
```
Comment (gtarpenning) [inline wandb/sdk/launch/agent/agent.py:267]:
```suggestion
            run_info = None
            try:
                run_info = self._api.get_run_info(
                    self._entity, job_and_run_status.project, job_and_run_status.run_id
                )
            except CommError:
                # normal runs have no info, sweep runs have run_info=None
                pass
                
           if not run_info:
                self.fail_run_queue_item(job_and_run_status.run_queue_item_id)
```

Comment (gtarpenning) [reply wandb/sdk/launch/agent/agent.py:267]: I feel like having self.fail_run_queue_item in only one place is slightly better

File: wandb/sdk/launch/agent/agent.py [hunk 4]
```diff
@@ -52,6 +57,12 @@ class JobAndRunStatus:
     def job_completed(self) -> bool:
         return self.completed or self.failed_to_start
 
+    def populate_run_info_from_launch_project(
```
Comment (gtarpenning) [inline wandb/sdk/launch/agent/agent.py:60]: nit
```suggestion
    def update_run_info(
```

Comment (gtarpenning) [reply wandb/sdk/launch/agent/agent.py:60]: we are passing in a type LaunchProject, easily inferred. 100000% up to you though.

**Review by gtarpenning — APPROVED** [review]
microscopic nits, did not actually test this branch with a sweep on launch, let me know if thats important.

**Review by raubitsj — APPROVED** [review]
LGTM for the new introspection

**Pull request merged**